helpers/plugins.js

```diff
@@ -0,0 +1,26 @@
+const pluginSets = new Map();
+
+export function getPlugin(setKey, pluginKey) {
+	const pluginSet = pluginSets.get(setKey);
+	return pluginSet?.plugins.find(plugin => plugin.options.key === pluginKey)?.plugin;
+}
+
+export function getPlugins(setKey) {
+	const pluginSet = pluginSets.get(setKey);
+	if (!pluginSet) return [];
+	if (pluginSet.requiresSorting) {
+		pluginSet.plugins.sort((item1, item2) => item1.options.sort - item2.options.sort);
+		pluginSet.requiresSorting = false;
+	}
+	return pluginSet.plugins.map(item => item.plugin);
+}
+
+export function registerPlugin(setKey, plugin, options) {
+	let pluginSet = pluginSets.get(setKey);
+	if (!pluginSet) {
+		pluginSet = { plugins: [], requiresSorting: false };
+		pluginSets.set(setKey, pluginSet);
+	}
```

Comment:
Nit: not sure if this is better or not, but another option that's 1 less line!

```javascript
if (!pluginSets.has(setKey)) {
  pluginSets.set({ plugins: [], requiresSorting: false });
}
const pluginSet = pluginSets.get(setKey);
```

Reply:
Oh yeah... I think I might had that initially, but then switched it to only do the single look-up. The difference is probably negligible for the number of plugin sets we'll have though.

Reply:
Should be `O(1)` lookup on the `Map` anyway.